vendorconsent/metadata20.go

```diff
@@ -0,0 +1,118 @@
+package vendorconsent
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/prebid/go-gdpr/consentconstants"
+)
+
+// Parse the metadata from the consent string.
+// This returns an error if the input is too short to answer questions about that data.
+func parseMetadata20(data []byte) (consentMetadata20, error) {
+	if len(data) < 30 {
+		return nil, fmt.Errorf("vendor consent strings are at least 30 bytes long. This one was %d", len(data))
+	}
+	metadata := consentMetadata20(data)
+	if metadata.MaxVendorID() < 1 {
+		return nil, fmt.Errorf("the consent string encoded a MaxVendorID of %d, but this value must be greater than or equal to 1", metadata.MaxVendorID())
+	}
+	if metadata.Version() < 1 {
+		return nil, fmt.Errorf("the consent string encoded a Version of %d, but this value must be greater than or equal to 1", metadata.Version())
+	}
+	if metadata.VendorListVersion() == 0 {
+		return nil, errors.New("the consent string encoded a VendorListVersion of 0, but this value must be greater than or equal to 1")
+
+	}
+	return consentMetadata20(data), nil
+}
+
+// consemtMetadata20 implements the parts of the VendorConsents interface which are common
+// to BitFields and RangeSections. This relies on Parse to have done some validation already,
+// to make sure that functions on it don't overflow the bounds of the byte array.
+type consentMetadata20 []byte
+
+func (c consentMetadata20) Version() uint8 {
+	// Stored in bits 0-5
+	return uint8(c[0] >> 2)
+}
+
+func (c consentMetadata20) Created() time.Time {
+	_ = c[5]
```

Comment:
Why is this line here?

Reply:
That is a good question. Its in the TFC 1 version but should be safe to remove.